src/System.Security.Cryptography.X509Certificates/tests/PublicKeyTests.cs

```diff
@@ -128,6 +129,56 @@ public static void TestKey_RSA()
         }
 
         [Fact]
+        public static void TestECDsaPublicKey()
+        {
+            byte[] helloBytes = Encoding.ASCII.GetBytes("Hello");
+
+            using (var cert = new X509Certificate2(TestData.ECDsa384Certificate))
+            using (ECDsa publicKey = cert.GetECDsaPublicKey())
+            {
+                Assert.Equal(384, publicKey.KeySize);
+
+                // The public key should be unable to sign.
+                Assert.ThrowsAny<CryptographicException>(() => publicKey.SignData(helloBytes, HashAlgorithmName.SHA256));
+            }
+        }
+
+        [Fact]
+        [ActiveIssue(3769, PlatformID.AnyUnix)]
+        public static void TestECDsaPublicKey_ValidatesSignature()
+        {
+            byte[] existingSignature =
+            {
+                // r:
```

Comment:
> where did we get this data blob from?

Attributed:
```
// This signature was produced as the output of ECDsaCng.SignData with the same key
// on .NET 4.6.  Ensure it is verified here as a data compatibility test.
//
// Note that since ECDSA signatures contain randomness as an input, this value is unlikely
// to be reproduced by another equivalent program.
```